### DISCUSSION TITLE
How to get an instance of the MainWindow in Avalonia?

### DISCUSSION THREAD

**Discussion by ApaulMyLittleAirport** [AvaloniaUI/Avalonia#8385] [Q&A] [answered]
wpf：
var _mainWindow = Application.Current.Windows
            .Cast<Window>()
            .FirstOrDefault(window => window is MainWindow) as MainWindow;

**Comment by FoggyFinder** [comment, marked as answer]
```fs
let window = self.GetVisualRoot() :?> Window
```

or

```fs
            match Application.Current.ApplicationLifetime with
            | :? IClassicDesktopStyleApplicationLifetime as desktop ->
                let window = desktop.MainWindow
            | _ -> ()
```

**Reply by ApaulMyLittleAirport** [reply]
> ```fsharp
> let window = self.GetVisualRoot() :?> Window
> ```
> 
> or
> 
> ```fsharp
>             match Application.Current.ApplicationLifetime with
>             | :? IClassicDesktopStyleApplicationLifetime as desktop ->
>                 let window = desktop.MainWindow
>             | _ -> ()
> ```

Nice!Successfully solved my problem！I appreciate your kindness and consideration.
